components/Posts.php

```diff
@@ -220,4 +252,20 @@ protected function checkEditor()
         $backendUser = BackendAuth::getUser();
         return $backendUser && $backendUser->hasAccess('rainlab.blog.access_posts');
     }
+
+    /**
+     * @param string $componentName
+     * @param string $page
+     * @return ComponentBase|null
+     */
+    protected function getComponent(string $componentName, string $page)
+    {
+        $component = Page::load(Theme::getActiveTheme(), $page)->getComponent($componentName);
+
+        if (!is_null($component)) {
+            $this->controller->setComponentPropertiesFromParams($component);
```

Comment:
Thinking about this, does this work even if those properties are not present in the current router object initialized by the current page URL? I.e. could this still work if this component is on a page with no URL params defined at all?

Reply:
Ok, here's my test setup: 

![image](https://user-images.githubusercontent.com/3897579/43047877-416ae4da-8dd6-11e8-9871-b45dbfdd82be.png)

![image](https://user-images.githubusercontent.com/3897579/43047902-be3dcd56-8dd6-11e8-9ca3-c3393294151d.png)

Sidebar includes `blogPosts` component for menu. The sidebar is included in every page and the welcome page has no parameters. Blog post page has ":post" parameter instead of default ":slug".

Anyway, the link is fine:

![image](https://user-images.githubusercontent.com/3897579/43047943-66a05ac2-8dd7-11e8-9c9e-2690bfe50d04.png)

Fine from the blog page too:

![image](https://user-images.githubusercontent.com/3897579/43047953-95454f9a-8dd7-11e8-8d09-f01cc7596312.png)

Reply:
Sweet, thanks for checking!